### ISSUE TITLE
About reproducing pretrain weight problems

### ISSUE THREAD

**Issue by ChenShiyuan-csc** [pkqbajng/CGFormer#18]
First of all, thank you for your great work! I really appreciate the efforts put into this project. However, I encountered some issues while running the pretraining experiment and would like to seek clarification.

1. When running the pretrain experiment, I noticed that the pretrained weight for the backone were not loaded. https://github.com/pkqbajng/CGFormer/blob/93bf428bcfed30a32b771b8b393a6f9e297345e6/configs/CGFormer-Efficient-Swin-SemanticKITTI-Pretrain.py#L143  Could this be due to an issue with my environment?
2. Based on the code logic, the validation step during pretraining does not compute the mIoU metric, resulting in an mIoU of 0 for all epochs. Since the checkpoint is saved based one the base mIoU, as seen in the code: https://github.com/pkqbajng/CGFormer/blob/93bf428bcfed30a32b771b8b393a6f9e297345e6/main.py#L56-L60 This means that best.ckpt corresponds to the first epoch's result. My own training results also confirm this behavior.
3. After resolving the first issue by manually loading the pretrained weights for the backbone and selecting the final epoch’s results to convert into my own efficientnet-seg-depth.pth, I used load_from to load my oefficientnet-seg-depth.pth. However, I was unable to reproduce your reported results, particularly in terms of mIoU. Could there be some differences in the pretrain configuration that I might have missed?

I would appreciate any insights or clarification regarding these discrepancies.
Thank you!

**Comment by pkqbajng** [comment]
For issue1, the pretrained checkpoint of efficientnet-b7_3rdparty_8xb32-aa_in1k_20220119-bf03951c.pth is indeed unnecessary, as we'll reload the checkpoint from the efficientnet-seg-depth.pth.

For issue2, we don't perform evaluation during the pretraining process, but the latest checkpoint is saved, please use this checkpoint.

After finish these two steps, it would be easy to reproduce the results in the paper.  The mIoU metric exhibits some fluctuation, approximately around 0.2.

**Comment by ChenShiyuan-csc** [comment]
Thank you for your reply!
I was able to successfully reproduce the final results using the checkpoint efficientnet-seg-depth.pth provided by you. However, I am currently looking to use the provided CGFormer-Efficient-Swin-SemanticKITTI-Pretrain.py to reproduce the pretraining stage. I assume that the efficientnet-b7_3rdparty_8xb32-aa_in1k_20220119-bf03951c.pth should be loaded during pretraining based on my understanding, but I’m a bit confused as to why the code doesn’t seem to load it.

**Comment by pkqbajng** [comment]
I got it. The checkpoint efficientnet-b7_3rdparty_8xb32-aa_in1k_20220119-bf03951c.pth should be loaded during pretraining process. There exists error when I opensourced this code, I'll update the overall project. Sorry for this.
